### ISSUE TITLE
getenv('PATH', A) has a 0-code replacing last character of A

### ISSUE THREAD

**Issue by anionic** [SWI-Prolog/issues#114]
For 64-bit Windows, starting with 8.5.11-1, we get e.g. (shortened)
```
?- getenv('PATH', Q).
Q = '...;C:\\Users\\paul\\AppData\\Local\\Microsoft\\WindowsApps\u0000'.
```
Doesn't matter what the last path component is, or exactly how long PATH is (haven't tried very short ones).
Other env vars (all much shorter) seem uncorrupted.

**Comment by JanWielemaker** [comment]
That doesn't reproduce, neither under Wine, nor under Windows 10.   Please try asking on Discourse and/or try to find something that I can reproduce.   Recently there was a change in the Windows binding to deal consistently with file name buffers.    That code is also used by the getenv() emulation, so yes, things did change.

**Comment by JanWielemaker** [comment]
Thanks.   Fixed with SWI-Prolog/swipl-devel@268c2849e282930711c5ab3886adae62633de5db.  Should be in tomorrows daily build.
